tests/unit/lib/package/test_artifact_exporter.py

```diff
@@ -1636,6 +1661,46 @@ def test_make_zip_windows(self, mock_system):
                 os.remove(zipfile_name)
             test_file_creator.remove_all()
 
+    def test_make_zip_lambda_resources(self):
```

Comment:
Can we have tests that verify each resource we are doing this to and another one for the unsupported ones? I would rather over test this to ensure no escalation for resources we don't intend too (now and the future).

At the same note, Can we have a test that also verifies local permissions are not escalated? The code looks correct to me but just want to have a good suite of tests surrounding this change.

Reply:
For every lambda resource: https://github.com/aws/aws-sam-cli/pull/4462/files#diff-756169fa4653144117b5d40b30eb21a08c4b4a19ab93d0029a0d69afa2e9a692R342 we test that we are only ever hitting `make_zip_with_lambda_permissions`